src/lib/Libutil/pbs_aes_encrypt.c

```diff
@@ -150,3 +152,101 @@ pbs_decrypt_pwd(char *crypted, int credtype, size_t len, char **uncrypted)
 
         return 0;
 }
+
+/**
+ * @brief
+ *	encode_to_base64 - Encode data into bas64 format
+ *
+ * @param[in]		buffer			-	Data buffer for encoding
+ * @param[in]		buffer_len		-	Length of the data buffer
+ * @param[in/out]	ret_encoded_data	-	Return the encoded data
+ *
+ * @return int
+ * @retval 0 - for success
+ * @retval 1 - for failure
+ */
+int
+encode_to_base64(const unsigned char* buffer, size_t buffer_len, char** ret_encoded_data)
+{
+	BUF_MEM *b_mem;
+	BIO *mem_obj1, *mem_obj2;
+	int buf_len = 0;
+
+	mem_obj1 = BIO_new(BIO_s_mem());
+	mem_obj2 = BIO_new(BIO_f_base64());
+
+	mem_obj1 = BIO_push(mem_obj2, mem_obj1);
+	BIO_set_flags(mem_obj1, BIO_FLAGS_BASE64_NO_NL);
+	BIO_write(mem_obj1, buffer, buffer_len);
+	(void)BIO_flush(mem_obj1);
+	BIO_get_mem_ptr(mem_obj1, &b_mem);
+
+	*ret_encoded_data = (*b_mem).data;
```

Comment:
Why are you doing this? In the next step you are using this pointer to malloc.

Reply:
yes, there is no sense of this step since we are using memcpy now. Thanks